modules/garbage-collection.adoc

```diff
@@ -0,0 +1,117 @@
+= {productname} garbage collection
+[[garbage-collection]]
+
+== About {productname} garbage collection
+
+{productname} includes automatic and continuous image garbage collection. Garbage collection ensures efficient use of resources for active objects by removing objects that occupy sizeable amounts of disk space, such as dangling or untagged images, repositories, and blobs, including layers and manifests. Garbage collection performed by {productname} can reduce downtime in your organization's environment.
+
+== {productname} garbage collection in practice
+
+Currently, all garbage collection happens discreetly, and there are no commands to manually run garbage collection. {productname} provides metrics that track the status of the different garbage collection workers.
+
+For namespace and repository garbage collection, the progress is tracked based on the size of their respective queues. Namespace and repository garbage collection workers require a global lock to work. As a result, and for performance reasons, only one worker runs at a time.
+
+Garbage collecting tagged images works differently than garbage collection on namespaces or repositories. Rather than having a queue of items to work with, the garbage collection workers for tagged images actively search for a repository with inactive or expired tags to clean up. Each instance of garbage collection workers will grab a repository lock, which results in one worker per repository. Unlike repository and namespace garbage collection, garbage collection for tagged images does not use a queue.
+
+For each type of garbage collection, {productname} provides metrics for the number of rows per table deleted by each garbage collection worker. The following image shows an example of how {productname} monitors garbage collection with the same metrics:
+
+image:garbage-collection-metrics.png[Garbage collection metrics]
+
+=== Measuring storage reclamation
+
+{productname} does not have a way to track how much space is freed up by garbage collection. Currently, the best sign of this is by checking how many blobs have been deleted in the provided metrics.
+
+[NOTE]
+====
+The `UploadedBlob` table in the {productname} metrics tracks the various blobs that are associated with a repository. When a blob is uploaded, it will not be garbage collected before the time designated by the `PUSH_TEMP_TAG_EXPIRATION_SEC` parameter. This is to avoid prematurely deleting blobs that are part of an ongoing push. For example, if garbage collection is set to run often, and a tag is deleted in the span of less than one hour, then it is possible that the associated blobs will not get cleaned up immediately. Instead, and assuming that the time designated by the `PUSH_TEMP_TAG_EXPIRATION_SEC` parameter has passed, the associated blobs will be removed the next time garbage collection runs on that same repository.
+====
```

Comment:
Good!!

Line 22: replace "sign" with "indicator."
Note: I think the content in the note goes closely enough with Line 22 that it should be outside a note.